backend/api/alerts.py

```
@@ -208,20 +88,50 @@ def get_alert_severities(credentials: HTTPAuthorizationCredentials = Depends(sec
-        cursor.execute("SELECT DISTINCT Severity FROM Alerts ORDER BY Severity")
-    else:
-        cursor.execute(
-            "SELECT DISTINCT Severity FROM Alerts WHERE SourceID = ? ORDER BY Severity",
-            (source_id,),
+@router.get("/getalertsbybatch")
+def get_alert_summary(
+    from_date: str,
+    to_date: str,
+    batches: Optional[str] = "",
+    severities: Optional[str] = "",
+    credentials: HTTPAuthorizationCredentials = Depends(security),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)